crates/trigger-http/src/wasi.rs

```diff
@@ -57,7 +58,18 @@ impl HttpExecutor for WasiHttpExecutor {
         )
         .context("missing OutboundHttpFactor")?;
 
-        let request = wasi_http.new_incoming_request(req)?;
+        let (parts, body) = req.into_parts();
+        let body = wasmtime_wasi_http::body::HostIncomingBody::new(
+            body,
+            std::time::Duration::from_secs(600),
+        );
+        let request = wasmtime_wasi_http::types::HostIncomingRequest::new(
+            &mut wasi_http,
+            parts,
+            Scheme::Http,
+            Some(body),
+        )?;
+        let request = wasi_http.table().push(request)?;
```

Comment:
Why are we manually creating the resource now instead of using `WasiHttpView::new_incoming_request`?

Reply:
Ah this is caused by a refactoring from a few versions ago in Wasmtime where the error type of the body is required to be `hyper::Error` for that helper but here it's `ErrorCode` from wasi-http instead. The Wasmtime helper will convert from `hyper::Error` to `ErrorCode` but there's not currently a constructor that works with `ErrorCode` itself. I tried to refactor Spin internals to use `hyper::Error` instead but ended up running into roadblocks around service chaining where `hyper::Error` was never materialized.

Perhaps the best solution would be to add a second helper or similar to the wasmtime-wasi-http crate though!